In [1]:
import pandas as pd
import numpy as np

In [2]:
#opening the file
df = pd.read_excel(r'/Users/anuliyanagamage/Downloads/Data_Challenge_SRPA.xlsx') 

In [3]:
# Dataframe
df

Brand Market  User_Id   Session_Id       Date   Device  \
0       Pitchfork   U.S.    10006  14483128093 2019-09-12  Desktop   
1       Pitchfork   U.S.    10006  14483128093 2019-09-12  Desktop   
2       Pitchfork   U.S.    10006  14483128093 2019-09-12  Desktop   
3       Pitchfork   U.S.    10006  14483128093 2019-09-12  Desktop   
4       Pitchfork   U.S.    10006  14483128093 2019-09-12  Desktop   
...           ...    ...      ...          ...        ...      ...   
375794       Self   U.S.    99996  22365367120 2019-07-27  Desktop   
375795       Self   U.S.    99996  54640177221 2019-08-17  Desktop   
375796       Self   U.S.    99996  54640177221 2019-08-17  Desktop   
375797       Self   U.S.    99996  68855396031 2019-11-13  Desktop   
375798       Self   U.S.    99996  68855396031 2019-11-13  Desktop   

        Event_Name Component_Name Site_Section  Event_Order  
0         Pageview        Article        Songs            1  
1       Impression          Video        Songs            2  
2       Impression  Advertisement        Songs            3  
3         Pageview        Gallery      Artists            4  
4       Impression          Video      Artists            5  
...            ...            ...          ...          ...  
375794  Impression  Advertisement      Fitness            5  
375795    Pageview        Gallery    Self Care            1  
375796  Impression  Advertisement    Self Care            2  
375797    Pageview        Gallery     Wellness            1  
375798  Impression  Advertisement     Wellness            2  

[375799 rows x 10 columns]

In [15]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 375799 entries, 0 to 375798
Data columns (total 10 columns):
 #   Column          Non-Null Count   Dtype         
---  ------          --------------   -----         
 0   Brand           375799 non-null  object        
 1   Market          375799 non-null  object        
 2   User_Id         375799 non-null  int64         
 3   Session_Id      375799 non-null  int64         
 4   Date            375799 non-null  datetime64[ns]
 5   Device          375799 non-null  object        
 6   Event_Name      375799 non-null  object        
 7   Component_Name  375799 non-null  object        
 8   Site_Section    370580 non-null  object        
 9   Event_Order     375799 non-null  int64         
dtypes: datetime64[ns](1), int64(3), object(6)
memory usage: 28.7+ MB


# How many Pageviews did Glamour have in August?

In [5]:
#converting date into year/month/day
df['Date'] = pd.to_datetime(df['Date'], format='%Y-%m-%d')

In [6]:
#filitering to month of auguest
August_df = df.loc[(df['Date'] >= '2019-07-30')
                     & (df['Date'] < '2019-09-01')]

In [7]:
#Finding pitchfork on auguest
Pitchfork_brand = August_df.loc[August_df['Brand'] == 'Glamour']

In [8]:
#Finding eventname Pageview
Glamour_brand_pageview = Pitchfork_brand.loc[Pitchfork_brand['Event_Name'] == 'Pageview']

In [9]:
Glamour_brand_pageview.count()

Brand             4355
Market            4355
User_Id           4355
Session_Id        4355
Date              4355
Device            4355
Event_Name        4355
Component_Name    4355
Site_Section      4355
Event_Order       4355
dtype: int64

In [10]:
Glamour_brand_pageview

Brand Market  User_Id   Session_Id       Date   Device Event_Name  \
76      Glamour   U.S.    10035  16360836369 2019-08-17  Desktop   Pageview   
78      Glamour   U.S.    10035  16360836369 2019-08-17  Desktop   Pageview   
81      Glamour   U.S.    10035  16717965079 2019-08-28  Desktop   Pageview   
409     Glamour   U.S.    10143  25868520446 2019-08-27  Desktop   Pageview   
411     Glamour   U.S.    10143  33750356728 2019-08-28   Mobile   Pageview   
...         ...    ...      ...          ...        ...      ...        ...   
374861  Glamour   U.S.    99780  15402031497 2019-08-06   Mobile   Pageview   
374995  Glamour   U.S.    99807  80294339729 2019-08-07   Mobile   Pageview   
374997  Glamour   U.S.    99807  80294339729 2019-08-07   Mobile   Pageview   
375398  Glamour   U.S.    99905  13818601891 2019-08-12  Desktop   Pageview   
375400  Glamour   U.S.    99905  13818601891 2019-08-12  Desktop   Pageview   

       Component_Name Site_Section  Event_Order  
76            Gallery     Shopping            1  
78            Gallery     Shopping            3  
81            Gallery     Shopping            1  
409           Article      Fashion            1  
411           Gallery     Shopping            1  
...               ...          ...          ...  
374861        Gallery     Shopping            1  
374995        Gallery       Beauty            1  
374997        Article       Beauty            3  
375398        Gallery       Beauty            1  
375400        Article       Beauty            3  

[4355 rows x 10 columns]

 * According to the dataset above Glamour brand had 4355 pageviews. 

# How many people saw a video on CNT?

In [24]:
# Finding CND brand
CNT_brand = df.loc[df['Brand'] == 'Conde Nast Traveler']

In [27]:
# Finding component video
CNT_brand_video = CNT_brand.loc[CNT_brand['Component_Name'] == 'Video']

In [41]:
#Dropped duplicate users who are on the same session
CNT_brand_video.drop_duplicates(subset=['Session_Id'])

Brand       Market  User_Id   Session_Id       Date  \
25      Conde Nast Traveler  Middle East    10011  13896085553 2019-09-29   
27      Conde Nast Traveler  Middle East    10011  24634628627 2019-10-11   
30      Conde Nast Traveler  Middle East    10011  89922012091 2019-06-26   
33      Conde Nast Traveler        Spain    10014  72438333891 2019-09-04   
37      Conde Nast Traveler        Spain    10014  80133001775 2019-08-30   
...                     ...          ...      ...          ...        ...   
375713  Conde Nast Traveler  Middle East    99968  92254512257 2019-11-02   
375716  Conde Nast Traveler  Middle East    99968  99904298145 2019-11-20   
375751  Conde Nast Traveler        Spain    99983  36142434745 2019-11-02   
375758  Conde Nast Traveler        Spain    99983  63534517045 2019-11-11   
375766  Conde Nast Traveler        Spain    99983  74433738770 2019-12-12   

         Device  Event_Name Component_Name   Site_Section  Event_Order  
25       Tablet  Impression          Video           Food            3  
27       Mobile  Impression          Video  Entertainment            2  
30      Desktop  Impression          Video           Food            2  
33      Desktop  Impression          Video       Homepage            2  
37      Desktop  Impression          Video           Food            3  
...         ...         ...            ...            ...          ...  
375713  Desktop  Impression          Video  Entertainment            2  
375716  Desktop  Impression          Video           Food            2  
375751  Desktop  Impression          Video  Entertainment            2  
375758  Desktop  Impression          Video  Entertainment            4  
375766  Desktop  Impression          Video         Sports            7  

[28020 rows x 10 columns]

In [42]:
#According to the data 45439 sessions were active to watch a video on CNT
CNT_brand_video.count()

Brand             45439
Market            45439
User_Id           45439
Session_Id        45439
Date              45439
Device            45439
Event_Name        45439
Component_Name    45439
Site_Section      45439
Event_Order       45439
dtype: int64

# #3 What percentage of SELF’s sessions are bounces. Which device type has the highest
bounce rate? We define a “bounce” as a session with only one pageview.


In [69]:
selff = df.loc[df['Brand'] == 'Self']

In [70]:
pageview = selff.loc[selff['Event_Name'] == 'Pageview']

In [71]:
pageview_Event_order = pageview.loc[pageview['Event_Order'] == 1]

In [81]:
# Desktops has the highest bounce rates
pageview_Event_order['Device'].value_counts()

Desktop    5616
Mobile     2860
Tablet      973
Name: Device, dtype: int64

In [96]:
selff['Event_Order'].value_counts()

1     9845
2     9845
3     7139
4     3908
5     2817
6     1695
7      841
8      458
9      250
10     116
11      58
12      28
13      13
14       5
15       2
16       1
Name: Event_Order, dtype: int64

In [114]:
# Calculating Percentage
selff["percent"] = (selff['Event_Order'] / selff['Event_Order'].sum()) * 100

<ipython-input-114-492cf250c7f7>:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  selff["percent"] = (selff['Event_Order'] / selff['Event_Order'].sum()) * 100


In [135]:
# Bounce Rate is 0.000952
selff["percent"].round(100)

145       0.000952
146       0.001904
147       0.002856
148       0.000952
149       0.001904
            ...   
375794    0.004759
375795    0.000952
375796    0.001904
375797    0.000952
375798    0.001904
Name: percent, Length: 37021, dtype: float64

In [138]:
# Bounce rates per session
selff

Brand Market  User_Id   Session_Id       Date   Device  Event_Name  \
145     Self   U.S.    10050  80866141615 2019-10-30  Desktop    Pageview   
146     Self   U.S.    10050  80866141615 2019-10-30  Desktop  Impression   
147     Self   U.S.    10050  80866141615 2019-10-30  Desktop  Impression   
148     Self   U.S.    10051  27453337199 2019-12-24   Mobile    Pageview   
149     Self   U.S.    10051  27453337199 2019-12-24   Mobile  Impression   
...      ...    ...      ...          ...        ...      ...         ...   
375794  Self   U.S.    99996  22365367120 2019-07-27  Desktop  Impression   
375795  Self   U.S.    99996  54640177221 2019-08-17  Desktop    Pageview   
375796  Self   U.S.    99996  54640177221 2019-08-17  Desktop  Impression   
375797  Self   U.S.    99996  68855396031 2019-11-13  Desktop    Pageview   
375798  Self   U.S.    99996  68855396031 2019-11-13  Desktop  Impression   

       Component_Name Site_Section  Event_Order   percent  
145           Article     Wellness            1  0.000952  
146             Video     Wellness            2  0.001904  
147             Video     Wellness            3  0.002856  
148           Article    Self Care            1  0.000952  
149             Video    Self Care            2  0.001904  
...               ...          ...          ...       ...  
375794  Advertisement      Fitness            5  0.004759  
375795        Gallery    Self Care            1  0.000952  
375796  Advertisement    Self Care            2  0.001904  
375797        Gallery     Wellness            1  0.000952  
375798  Advertisement     Wellness            2  0.001904  

[37021 rows x 11 columns]